src/components/app/details/cdDetails/deploymentHistoryDiff/DeploymentHistoryDetailedView.tsx

```diff
@@ -0,0 +1,114 @@
+import { getDeploymentHistoryDetail, prepareHistoryData } from '../service'
+import { useParams } from 'react-router'
+import {
+    DeploymentTemplateOptions,
+    CompareViewDeploymentType,
+    DeploymentHistoryDetail,
+    DeploymentHistoryParamsType,
+} from '../cd.type'
+import DeploymentHistorySidebar from './DeploymentHistorySidebar'
+import DeploymentHistoryDiffView from './DeploymentHistoryDiffView'
+
+export default function DeploymentHistoryDetailedView({
+    showTemplate,
+    setShowTemplate,
+    loader,
+    setLoader,
+    deploymentHistoryList,
+    setDeploymentHistoryList,
+}: CompareViewDeploymentType) {
+    const { appId, pipelineId, historyComponent, baseConfigurationId, historyComponentName } =
+        useParams<DeploymentHistoryParamsType>()
+    const [selectedDeploymentTemplate, setSelectedDeploymentTemplate] = useState<DeploymentTemplateOptions>()
+    const [currentConfiguration, setCurrentConfiguration] = useState<DeploymentHistoryDetail>()
+    const [baseTemplateConfiguration, setBaseTemplateConfiguration] = useState<DeploymentHistoryDetail>()
+    const [codeEditorLoading, setCodeEditorLoading] = useState<boolean>(false)
+    const [previousConfigAvailable, setPreviousConfigAvailable] = useState<boolean>(true)
+
+    useEffect(() => {
+        setLoader(true)
+
+        if (selectedDeploymentTemplate) {
+            try {
+                getDeploymentHistoryDetail(
+                    appId,
+                    pipelineId,
+                    selectedDeploymentTemplate.value,
+                    historyComponent,
+                    historyComponentName,
+                ).then((response) => {
+                    setCurrentConfiguration(prepareHistoryData(response.result, historyComponent))
+                    setLoader(false)
+                })
+            } catch (err) {
+                showError(err)
+                setLoader(false)
+            }
+        }
+    }, [selectedDeploymentTemplate])
+
+    useEffect(() => {
+        try {
+            setCodeEditorLoading(true)
+            if (baseConfigurationId) {
+                getDeploymentHistoryDetail(
+                    appId,
+                    pipelineId,
+                    baseConfigurationId,
+                    historyComponent,
+                    historyComponentName,
+                ).then((response) => {
+                    setBaseTemplateConfiguration(prepareHistoryData(response.result, historyComponent))
+                    setCodeEditorLoading(false)
+                })
+            }
+        } catch (err) {
+            showError(err)
+            setCodeEditorLoading(false)
+        }
+    }, [baseConfigurationId, historyComponent, historyComponentName])
+
+    useEffect(() => {
+        if (!showTemplate) {
+            setShowTemplate(true)
+        }
+
+        return (): void => {
+            if (showTemplate) {
+                setShowTemplate(false)
+            }
+        }
+    }, [showTemplate])
```

Comment:
We just need to set this state on the mount/load so remove the `showTemplate` from the array. 
Also, add a comment.